R/db_create_dsm_table.R

```diff
@@ -122,20 +122,17 @@ db_create_dsm_table <- function(lcu_table,
   # Add is_interpolated column
   dt$is_interpolated <- FALSE
 
+  # Add is_used_for_line_up column
+  cc <- check_no_national_survey(dt) # Check for no national surveys
+  dt[, n_rl := .N, by = cache_id]    # Create number of rows per cache_id
+  check <- (dt$reporting_level == "national" & dt$n_rl == 1) |  # Surveys w/ national reporting level and no split by U/R domain (e.g USA)
+    (dt$reporting_level %in% c("urban", "rural") & dt$n_rl == 2) | # Surveys split by U/R domain (e.g. CHN, IND)
+    dt$country_code %in% cc  # Countries wo/ any national surveys (e.g. ARG, SUR)
+  dt[, is_used_for_line_up := ifelse(check, TRUE, FALSE)]
+
   # Add is_used_for_aggregation column
-  # Temporary quick fix for is_used_for_aggregation column,
-  # see issue PIP-Technical-Team/TMP_pipeline#14
-
-  dt[, # create number of rows per cache_id
-     n_rl := .N,
-     by = cache_id
-     ][,
-       # variable for aggregate
-       is_used_for_aggregation := fifelse(n_rl > 1, TRUE, FALSE)
-     ][,
-       # remove counter
-       n_rl := NULL
-     ]
+  dt[, is_used_for_aggregation := ifelse((dt$reporting_level %in% c("urban", "rural") & dt$n_rl == 2), TRUE, FALSE)]
+  dt$n_rl <- NULL
```

Comment:
I feel that this code chunk (lines 127 to 132) should live in its own function and be unit tested. This is an important piece of the PIP logic, and it will be easier to manage and document in a self-contained function.

Reply:
Okay. I can try to add a function and corresponding test for this.

Reply:
Added. Let me know what you think